### PR TITLE
Correction for Rolling CI testing failling due to change in ros2control (base_interface)

### PR DESCRIPTION
**Pull request by BenjaminHug8**
This PR is to solve the romve of `base_interface` from `hardware_interface` from `ros2control` as it makes the rolling testing test fail on CI.

Also it impacts `ROS Galactic`.

This take work from @Timple from this PR #313 and insure compatibility with `foxy`.

### CODE REVIEW

File: webots_ros2_control/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,17 @@
 cmake_minimum_required(VERSION 3.5)
 project(webots_ros2_control)
 
+# Check which ROS distribution is used, ros2control depends of that
```
Comment (Timple) [inline webots_ros2_control/CMakeLists.txt:4]: Would it make sense to add galactic?

File: webots_ros2_control/include/webots_ros2_control/Ros2Control.hpp [hunk 2]
```diff
@@ -20,12 +20,15 @@
 #include <vector>
 #include <thread>
 
-#include "hardware_interface/base_interface.hpp"
+#if FOXY || (ROLLING && MAIN)
```
Comment (Timple) [inline webots_ros2_control/include/webots_ros2_control/Ros2Control.hpp:23]: I am assuming that `rolling` and `galactic` have the same breaking changes. So `foxy` would be the exception?

File: webots_ros2_control/CMakeLists.txt (hunk 1 above)
Comment (BenjaminHug8) [reply webots_ros2_control/CMakeLists.txt:4]: Yes we can, but as the code is written there is changes when we use `foxy` or `rolling` main release.

Thus in any other case the code will be as for `rolling` testing release (same as `Galactic`).

File: webots_ros2_control/include/webots_ros2_control/Ros2Control.hpp (hunk 2 above)
Comment (BenjaminHug8) [reply webots_ros2_control/include/webots_ros2_control/Ros2Control.hpp:23]: Yes exactly!

**Comment by BenjaminHug8** [comment]
It seems from [this PR](https://github.com/ros-controls/ros2_control/pull/566) that some work is going to be published to allow `import_component()` to initialize the system and I believe we will benefit from that, thus I am going to wait for that.

**Comment by omichel** [comment]
It seems that galactic testing is failing in the same way rolling testing, while galactic main is failing because of missing packages.

**Comment by BenjaminHug8** [comment]
> It seems from [this PR](https://github.com/ros-controls/ros2_control/pull/566) that some work is going to be published to allow `import_component()` to initialize the system and I believe we will benefit from that, thus I am going to wait for that.

PR has been merged yesterday evening, going to test this morning.

**Comment by BenjaminHug8** [comment]
It is working with `rolling testing` but not for `galacting testing`, I am working on it.

**Comment by BenjaminHug8** [comment]
`galacting testing` is working too now!

**Comment by BenjaminHug8** [comment]
Seems that `rolling testing` has been merged in `rolling main`, going to adapt that.

Otherwise for `rolling testing` and `galactic testing` we depend on the release of `ros2_control`.

**Review by omichel — APPROVED** [review]
Let's hope the fix for ros2_control will be released soon.

**Pull request merged**